src/CabanaPD_Input.hpp

```diff
@@ -12,102 +12,73 @@
 #ifndef INPUTS_H
 #define INPUTS_H
 
+#include <fstream>
+#include <iostream>
 #include <string>
 
-#include <CabanaPD_Output.hpp>
+#include <nlohmann/json.hpp>
 
 namespace CabanaPD
 {
-template <int Dim = 3>
 class Inputs
 {
   public:
-    std::string output_file = "cabanaPD.out";
-    std::string error_file = "cabanaPD.err";
-    std::string device_type = "SERIAL";
+    Inputs( const std::string filename )
+    {
+        // Get user inputs.
+        inputs = parse( filename );
 
-    std::array<int, Dim> num_cells;
-    std::array<double, Dim> low_corner;
-    std::array<double, Dim> high_corner;
+        // Add additional derived inputs to json. System size.
+        auto size = inputs["system_size"]["value"];
+        for ( std::size_t d = 0; d < size.size(); d++ )
+        {
+            double s = size[d];
+            inputs["low_corner"]["value"][d] = -0.5 * s;
```

Comment:
Should probably add units based on the system size units

Reply:
Units added.